lib/shoulda/matchers/active_model/validate_comparison_of_matcher.rb

```diff
@@ -0,0 +1,587 @@
+      #     class PersonTest < ActiveSupport::TestCase
+      #       should validate_comparison_of(:favorite_color).is_equal_to("blue")
+      #     end
+      #
+      #
+      # ##### is_other_than
+      #
+      # Use `is_other_than` to test usage of the `:other_than` option.
+      # This asserts that the attribute can take a number which is not equal to
+      # the given value.
+      #
+      #     class Person
+      #       include ActiveModel::Model
+      #       attr_accessor :legal_age
+      #
+      #       validates_comparison_of :legal_age, other_than: 21
+      #     end
+      #
+      #     # RSpec
+      #     RSpec.describe Person, type: :model do
+      #       it do
+      #         should validate_comparison_of(:legal_age).
+      #           is_other_than(21)
+      #       end
+      #     end
+      #
+      #     # Minitest (Shoulda)
+      #     class PersonTest < ActiveSupport::TestCase
+      #       should validate_comparison_of(:legal_age).
+      #         is_other_than(21)
+      #     end
+      #
+      # ##### with_message
+      #
+      # Use `with_message` if you are using a custom validation message.
+      #
+      #     class Person
+      #       include ActiveModel::Model
+      #       attr_accessor :number_of_dependents
+      #
+      #       validates_comparison_of :number_of_dependents, greater_than: 0
+      #         message: 'Number of dependents must be a number'
+      #     end
+      #
+      #     # RSpec
+      #     RSpec.describe Person, type: :model do
+      #       it do
+      #         should validate_comparison_of(:number_of_dependents).
+      #           is_greater_than(0).
+      #           with_message('Number of dependents must be a number')
+      #       end
+      #     end
+      #
+      #     # Minitest (Shoulda)
+      #     class PersonTest < ActiveSupport::TestCase
+      #       should validate_comparison_of(:number_of_dependents).
+      #         is_greater_than(0).
+      #         with_message('Number of dependents must be a number')
+      #     end
+      #
+      # ##### allow_nil
+      #
+      # Use `allow_nil` to assert that the attribute allows nil.
+      #
+      #     class Post
+      #       include ActiveModel::Model
+      #       attr_accessor :age
+      #
+      #       validates_comparison_of :age, greater_than: 0, allow_nil: true
+      #     end
+      #
+      #     # RSpec
+      #     RSpec.describe Post, type: :model do
+      #       it { should validate_comparison_of(:age).is_greater_than(0).allow_nil }
+      #     end
+      #
+      #     # Minitest (Shoulda)
+      #     class PostTest < ActiveSupport::TestCase
+      #       should validate_comparison_of(:age).is_greater_than(0).allow_nil
+      #     end
+      #
+      # @return [ValidateComparisonOfMatcher]
+      #
+      def validate_comparison_of(attr)
+        ValidateComparisonOfMatcher.new(attr)
+      end
+
+      # @private
+      class ValidateComparisonOfMatcher
```

Comment:
So, one quirk of ValidateNumericalityOfMatcher (which is where I presume you copied this code from) is that it doesn't inherit from ValidationMatcher, whereas the other matchers do. It's something that we've been wanting to fix for a long time but have just never gotten around to it. I'd like to not continue that pattern if we can 😂 That said, I know that there are a lot of extra things in here that aren't in ValidationMatcher. How complicated in your estimation is it to use ValidationMatcher instead? If it's too much trouble then perhaps we can refactor this in a future PR but just wanted to point that out.

Reply:
It shouldn't be too complicated, but I'll have to look deeply to ensure. But I can and would be happy to work on a follow-up PR for refactoring that. That PR is already big and complex, and I would like to avoid adding more complexity to it if that's fine for you.

Reply:
Yup, no problem!